File: examples/android/MSDKSampleAppJava/app/src/main/java/msdk/java/sample/history/HistoryViewModel.java
package msdk.java.sample.history;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;
import java.util.concurrent.Executors;

import msdk.java.sample.SingleLiveData;
import msdk.java.sample.db.Database;
import msdk.java.sample.db.entity.Action;
import msdk.java.sample.homepage.Results;

import static msdk.java.sample.db.ActionStatus.HISTORIZED;
import static msdk.java.sample.homepage.Results.ACTION_FAILURE;
import static msdk.java.sample.homepage.Results.ACTION_SUCCESS;
import static msdk.java.sample.homepage.Results.FAILURE;
import static msdk.java.sample.homepage.Results.REJECT;

public class HistoryViewModel extends AndroidViewModel {
    private final Database db;
    private LiveData<List<Action>> actions;

    public HistoryViewModel(@NonNull Application application) {
        super(application);
        db = Database.getInstance(application);
    }

    public LiveData<List<Action>> getHistory() {
        if (actions == null) {
            actions = db.actionDao().getActionsByStatus(HISTORIZED.toString());
        }
        return actions;
    }
}

File: examples/android/MSDKSampleAppKotlin/app/src/main/java/msdk/kotlin/sample/history/HistoryViewModel.kt
package msdk.kotlin.sample.history

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import msdk.kotlin.sample.db.Database
import msdk.kotlin.sample.db.entity.Action

import msdk.kotlin.sample.db.ActionStatus.HISTORIZED

class HistoryViewModel(application: Application) : AndroidViewModel(application) {
    private var db = Database.getInstance(application)
    private val actionsLiveData: LiveData<List<Action>> by lazy {
        db.actionDao().getActionsByStatus(HISTORIZED.toString())
    }

    fun getHistory(): LiveData<List<Action>> = actionsLiveData
}